modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,897 @@
+        type: str
+        description:
+        - Base snapshot name. This attribute is required together with name and clone when cloning a volume with the create operation.
+    volcoll:
+        required: False
+        type: str
+        description:
+        - Name of volume collection of which this volume is a member.
+    force:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly offline, reduce size or change read-only status a volume.
+    caching_enabled:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Indicate caching the volume is enabled.
+    force_vvol:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly move a Virtual volume.
+    move:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Move a volume to different Pool.
+
+extends_documentation_fragment: hpe_nimble
+short_description: "Manages a HPE Nimble Storage volume"
+version_added: "2.9"
+'''
+
+EXAMPLES = r'''
+
+    # If state is "create", then create a volume if not present. Fails if already present.
+    # If state is "present", then create a volume if not present. Succeeds if it already exists.
+    - name: Create volume if not present
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: "{{ state | default('present') }}" # fail if exist
+        size: "{{ size }}"
+        limit_iops: "{{ limit_iops }}"
+        limit_mbps: 5000
+        force: false
+        metadata: "{{ metadata }}"
+        description: "{{ description }}"
+        name: "{{ name }}"
+
+    - name: Changing volume "{{ name }}" to offline state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: offline
+        name: "{{ name }}"
+
+    - name: Changing volume "{{ name }}" to online state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: online
+        name: "{{ name }}"
+
+    # Create a clone from the given snapshot name.
+    # If snapshot name is not provided then a snapshot is created on the source volume.
+    # Clone task only run if "parent" is specified. Snapshot is optional.
+    - name: Create or Refresh a clone!
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}" # name here is the name of cloned volume
+        parent: "{{ parent | mandatory }}"
+        snapshot: "{{ snapshot | default(None)}}"
+        state: "{{ state | default('present') }}" # fail if exist
+      when:
+        - parent is defined
+
+    - name: Destroy volume (it's not offline)
```

Comment:
What does this mean? Shouldn't it read "must be offline"?

Reply:
yes. changed